src/main/java/gov/usgs/owi/nldi/controllers/BaseController.java

```diff
@@ -94,6 +94,35 @@ protected void streamFlowLines(HttpServletResponse response,
 		streamResults(transformer, featureType, parameterMap);
 	}
 
+	protected void streamFlowLines(HttpServletResponse response,
+			String comid, String navigationMode, String stopComid, String distance, String measure, String trimTolerance, boolean legacy) throws Exception {
+
+		String featureType;
+		Map<String, Object> parameterMap = parameters.processParameters(comid, navigationMode, distance, stopComid, measure, trimTolerance);
+
+		if (!parameterMap.isEmpty() && parameterMap.get(Parameters.TRIM_TOLERANCE) == null) {
+			parameterMap.put(Parameters.TRIM_TOLERANCE, Parameters.TRIM_TOLERANCE_DEFAULT);
+		}
+
+		if (legacy) {
```

Comment:
Ignore this if it's not relevant but is there some kind of versioning mechanism instead of adding a legacy flag argument here? Seems like this could leave around old (legacy) stuff that really should just be phased out or left on an older version.

Reply:
I'm unsure how much/how long we plan to have legacy request support for. I agree that it should eventually be phased out. @dblodgett-usgs might have further input.